libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -402,36 +402,34 @@ - (NSString *)sfHybridViewUserAgentString
     return userAgentString;
 }
 
-- (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrl returnUrlIsEncoded:(BOOL)isEncoded createAbsUrl:(BOOL)createAbsUrl
+- (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrlString returnUrlIsEncoded:(BOOL)isEncoded createAbsUrl:(BOOL)createAbsUrl
 {
     SFOAuthCredentials *creds = [SFUserAccountManager sharedInstance].currentUser.credentials;
-    NSString *instUrl = creds.apiUrl.absoluteString;
-    NSString *fullReturnUrl = returnUrl;
-
+    NSURL *instUrl = creds.apiUrl;
+    NSString *fullReturnUrlString = returnUrlString;
+    
     /*
      * We need to use the absolute URL in some cases and relative URL in some
      * other cases, because of differences between instance URL and community URL.
      */
-    if (createAbsUrl && ![returnUrl hasPrefix:@"http"]) {
-        fullReturnUrl = [NSString stringWithFormat:@"%@/%@", instUrl, returnUrl];
+    if (createAbsUrl && ![returnUrlString hasPrefix:@"http"]) {
+        fullReturnUrlString = [instUrl URLByAppendingPathComponent:returnUrlString].absoluteString;
```

Comment:
The former version was subject to blindly creating URLs like `http://example.com/root//subtree`.  I pulled back to a standard convention for appending paths.

Reply:
`URLByAppendingPathComponent` has some issues with base URLs that have more than one path component such as custom community URLs. We used to do this in the past and I changed it to string appending to fix those bugs.

Reply:
Ideally, it would be nice if we could build up a string all the way through in this method.

Reply:
OMG that's crazy stupid, to the point of not even understanding how my change actually worked in my use case.  I created a test to go through various scenarios, and `URLByAppendingPathComponent` is nearly useless. 🙁

Okay, let me take a look at "manually" bullet proofing the scenario I was seeing.  I'll also look based on that and see if the structure of the builder functionality can be cleaned up a bit.